### ISSUE TITLE
postInstallation Fails - Duplicate entry '1' for key 'PRIMARY' - 2020.7.0-debian-10-r52

### ISSUE THREAD

**Issue by jimmybrancaccio** [bitnami/bitnami-docker-phabricator#142]
**Description**

This appears to be the same issue in #131 however I am not sure how to get it resolved as I already have a database in MariaDB for Phabricator. This database has been in use for a while now, and hasn't had any problems being used when new versions of the Bitnami Phabricator image have been released. I am not positive what query is failing or else I could provide more information.

**Steps to reproduce the issue:**

1. Have an already running instance of Phabricator using the Bitnami Docker image.
2. Follow the upgrade steps as outlined in this projects README using 2020.7.0-debian-10-r52.
3. Start up your container again, and review the logs.

**Describe the results you received:**

Looking at the containers logs, I see:

```
Welcome to the Bitnami phabricator container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-phabricator
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-phabricator/issues
Send us your feedback at containers@bitnami.com

nami INFO Initializing apache
nami INFO apache successfully initialized
nami INFO Initializing php
nami INFO php successfully initialized
nami INFO Initializing phabricator
phabric INFO Creating Phabricator system users...
phabric INFO Creating Phabricator temporary folders...
phabric INFO Configuring PHP settings...
phabric INFO Configuring Apache settings...
phabric INFO Checking connection to database...
mysql-c INFO Trying to connect to MySQL server
mysql-c INFO Found MySQL server listening at 172.18.0.10:3306
mysql-c INFO MySQL server listening and working at 172.18.0.10:3306
phabric INFO Creating Database...
phabric INFO Configuring Phabricator...
phabric INFO Configuring SMTP...
Error executing 'postInstallation': MySQL command failed to run. Error:
ERROR 1062 (23000) at line 1: Duplicate entry '1' for key 'PRIMARY'
```

**Describe the results you expected:**

I expect the container would not attempt to create a new database, but just run the upgrade/sql migrations. It might be trying to in fact, but I am not sure. Either way it fails.

**Additional information you deem important (e.g. issue happens only occasionally):**

I've removed the Phabricator container and started fresh. I deleted the Bitnami Phabricator image and re-pulled it, but nothing seems to work, it just fails each time. I use Docker Compose, but that has all been working fine as well, nothing has really changed in my setup since it was last working.

I wish I knew what this 'postInstall' was doing exactly, but I can't seem to figure it out. If any further details are needed, please don't hesitate to let me know!

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           19.03.6
 API version:       1.40
 Go version:        go1.12.16
 Git commit:        369ce74a3c
 Built:             Thu Feb 13 01:27:49 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.6
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.16
  Git commit:       369ce74a3c
  Built:            Thu Feb 13 01:26:23 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 47
  Running: 41
  Paused: 0
  Stopped: 6
 Images: 78
 Server Version: 19.03.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.3.0-46-generic
 Operating System: Ubuntu 19.10
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 31.36GiB
 Name: yoshi
 ID: PKRO:LFYL:KH4C:JQ76:A445:WHHR:5EWF:5AIE:AYP5:Z65K:5A4K:7RJH
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: jimmyb
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: API is accessible on http://0.0.0.0:2375 without encryption.
         Access to the remote API is equivalent to root access on the host. Refer
         to the 'Docker daemon attack surface' section in the documentation for
         more information: https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface
WARNING: No swap limit support
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.25.0, build 0a186604
docker-py version: 4.1.0
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by juan131** [comment]
Hi @jimmybrancaccio 

It seems Phabricator is trying to initialize your database from scratch instead of reusing the data. As you can see in the upgrade instructions (check the link below), you should make snapshot of both MariaDB and Phabricator data volumes, so you can go back to the previous version if sth goes wrong:

- https://github.com/bitnami/bitnami-docker-phabricator#upgrading-phabricator

Did you persist the data in the Phabricator container as explained in the link below? The 1s time Phabricator container is launched, it creates a flag at `/bitnami/phabricator` directory that should be persisted. That flags indicates that on future restarts of the container, it doesn't need to populate the database but reuse its data upgrading the schema instead. According to your logs, Phabricator doesn't find that flag and, therefore, tries to populate the database from scractch.

- https://github.com/bitnami/bitnami-docker-phabricator#persisting-your-application

**Comment by jimmybrancaccio** [comment]
Hey @juan131! What's the file/flag it's checking for to see if it should create a DB or not? It's all been working for months now so I am not sure what happened.

Actually, I think I see which one it might be -- `.initialized`?

I have a feeling it's the permissions being all messed up?

```
[jimmy@yoshi.linuxbox.ninja ~/.docker/config/phabricator]$ ls -la
total 32
drwxr-xr-x  8 root   root   4096 Apr 13 12:09 .
drwxr-xr-x 33 jimmy  jimmy  4096 Apr 19 23:12 ..
drwxr-xr-x  2 root   root   4096 Apr 13 12:09 .bash_history
drwxr-xr-x  5 root   root   4096 Jan 24 00:30 conf
drwxr-xr-x 86 daemon daemon 4096 Mar 24 15:08 data
-rw-r--r--  1 root   root      0 Jan 24 00:30 .initialized
drwxr-xr-x  3 root   root   4096 Apr 13 12:09 phabricator
drwxr-xr-x  3   1001 docker 4096 Jan 25 13:10 repo
-rw-r--r--  1 root   root      0 Apr 13 05:00 .restored
drwxr-xr-x  2 root   root   4096 Jan 23 23:12 .sshkeys
```

What should these permissions be? Most of my Docker containers run as my user, but it seems this Phabricator container may not support this?

**Comment by jimmybrancaccio** [comment]
Ok, I think I got it all back and working. Something must have either changed in how this deploys.

I moved most everything listed above into my `~/.docker/config/phabricator/phabricator` directory. I also had to run a `mysql_upgrade` on my MariaDB 10.4 server, and I reset password for the MySQL user `bn_phabricator` (just used the same one that was in `local.js`). I removed and recreated the container and then it seemed happy. I also fixed up the permissions on the directories while I was at it so it was all owned/grouped to the phabricator user.
